internal/controller/policyexceptiondraft_controller.go

```diff
@@ -69,118 +76,164 @@ func (r *PolicyExceptionDraftReconciler) Reconcile(ctx context.Context, req ctrl
+	policyException.Labels["app.kubernetes.io/managed-by"] = "kyverno-policy-operator"
+	// Set ownerReferences
+	if err := controllerutil.SetControllerReference(&exceptionDraft, &policyException, r.Scheme); err != nil {
+		return ctrl.Result{}, err
+	}
 
-		// Create Kyverno exception
-		// Translate GiantSwarm PolicyExceptionDraft to Kyverno's PolicyException schema
-		policyException := translateDraftToPolex(exceptionDraft)
-		// Set namespace
-		policyException.Namespace = namespace
-		// Set name
-		policyException.Name = exceptionDraft.Name
-		// Set labels
-		policyException.Labels = make(map[string]string)
-		policyException.Labels["app.kubernetes.io/managed-by"] = "kyverno-policy-operator"
-		// Set Background behaviour
-		policyException.Spec.Background = &background
-
-		// Set ownerReferences
-		if err := controllerutil.SetControllerReference(&exceptionDraft, &policyException, r.Scheme); err != nil {
-			return ctrl.Result{}, err
-		}
+	// Create PolicyException
+	if op, err := controllerutil.CreateOrUpdate(ctx, r.Client, &policyException, func() error {
 
-		// Create PolicyException
-		if err := r.Create(ctx, &policyException); err != nil {
-			if apierrors.IsAlreadyExists(err) {
-				log.Log.Info(fmt.Sprintf("PolicyException %s/%s already exists", namespace, policyException.Name))
-				return ctrl.Result{}, nil
-			} else {
-				log.Log.Error(err, "unable to create PolicyException")
-			}
-		} else {
-			log.Log.Info(fmt.Sprintf("Created PolicyException %s/%s", namespace, policyException.Name))
+		// Set .Spec.Match.Any targets
+		policyException.Spec.Match.Any = translateTargetsToResourceFilters(exceptionDraft)
+
+		// Set .Spec.Exceptions
+		newExceptions := translatePoliciesToExceptions(policyMap)
+		if !deepEquals(policyException.Spec.Exceptions, newExceptions) {
+			policyException.Spec.Exceptions = newExceptions
 		}
+
+		return nil
+	}); err != nil {
+		log.Log.Error(err, fmt.Sprintf("Reconciliation failed for PolicyException %s", policyException.Name))
+		return ctrl.Result{}, err
 	} else {
-		// Exception must exist since draft was previously reconciled
-		var policyException kyvernov2alpha1.PolicyException
+		log.Log.Info(fmt.Sprintf("PolicyException %s: %s", policyException.Name, op))
+	}
 
-		if err := r.Get(ctx, types.NamespacedName{Namespace: namespace, Name: exceptionDraft.Name}, &policyException); err != nil {
-			// Error fetching the report
-			if apierrors.IsNotFound(err) {
-				// The PolEx doesn't exist, remove the reconciled label from the PolicyExceptionDraft to trigger recreation
-				delete(exceptionDraft.Labels, "kyverno-policy-operator/reconciled")
+	return ctrl.Result{}, nil
+}
 
-				// Update Kubernetes object
-				if draftErr := r.Client.Update(ctx, &exceptionDraft, &client.UpdateOptions{}); draftErr != nil {
-					r.Log.Error(draftErr, "unable to update PolicyExceptionDraft")
+func deepEquals(got []kyvernov2alpha1.Exception, want []kyvernov2alpha1.Exception) bool {
```

Comment:
Have you tried any other `deepEquals` functions? Might not need to roll your own here